### ISSUE TITLE
Support release syncing to CentOS Stream

### ISSUE THREAD

**Issue by lbarcziova** [packit/packit-service#2068]
Followup of packit/packit#1979:

- [ ] update the [parsing of the message from Upstream release monitoring](https://github.com/packit/packit-service/blob/38d6158d1b718a079d4b08c35bbd69caf2e7859b/packit_service/worker/parser.py#L1484) if needed
   - [here](https://github.com/packit/packit-service/blob/38d6158d1b718a079d4b08c35bbd69caf2e7859b/packit_service/worker/parser.py#L1501) we construct the URL of the package in Fedora dist-git, this needs to be done for CentOS Stream (the repo for the particular package in [CentOS Stream dist-git on GitLab](https://gitlab.com/redhat/centos-stream/rpms)) as well
   - this URL should be used as well [when looking](https://github.com/packit/packit-service/blob/38d6158d1b718a079d4b08c35bbd69caf2e7859b/packit_service/worker/events/new_hotness.py#L79) for the configuration (of pull-from-upstream)
- [ ] update the [call of sync_release](https://github.com/packit/packit-service/blob/38d6158d1b718a079d4b08c35bbd69caf2e7859b/packit_service/worker/handlers/distgit.py#L214) according to the implementation in packit/packit#1979 => for `dist_git_branches` such as c8s, c9s trigger the release sync in CentOS Stream and not Fedora
---
- [ ] the retriggering should be later possible as well, but it is currently being worked on in #1829 (will update the action items once the implementation is done)

CC @cathay4t

**Comment by mfocko** [comment]
I had a look at it… It is a bit more complicated to resolve:

* parsing is not affected that much, yet
  * we should make sure we're using the correct package name for both Fedora and CentOS
  * we should include the CentOS dist-git
* handling of the CentOS _v._ Fedora should be separated, since they depend on the dist-git included in the API and we can have only one dist-git, duh…
  * having a separate “package” (one for Fedora, one for CentOS) in Packit config might ease this a lot, since there are different `dist_git_branches` and each “package” would have its own downstream counterpart, there would be 2 separate Celery tasks for handling this, thus resolving the `PackitAPI` issue
* given the two points above, it would probably make sense to extend the event to hold dist-git URL**s** instead of just the one

**Comment by lbarcziova** [comment]
The person who will pick this should make sure this is everything that needs to be changed and in case there is more, create follow-up issues.

**Comment by nforro** [comment]
I wonder if this actually makes sense. We react to `org.fedoraproject.prod.hotness.update.bug.file` messages, which effectively means [release-monitoring.org](https://release-monitoring.org) projects that have Fedora mapping configured and monitoring enabled at [src.fedoraproject.org](https://src.fedoraproject.org). Shouldn't we rather react to `org.release-monitoring.prod.anitya.project.version.update` messages and check for Fedora/CentOS mapping and continue from there?

**Comment by lachmanfrantisek** [comment]
@nforro Thanks for bringing this up.

If I remember correctly, we've picked this option to have some pre-filtering done by the New Hotness but I am now not sure, what pre-filtering is being done here and if it still makes sense. We can ask Michal.

**Comment by nforro** [comment]
Yes, I suppose the monitoring option in Pagure does filter out some unwanted traffic, and it makes complete sense to take advantage of it in Fedora, but not so much in CentOS. What if there is a package that is EL-only?

**Comment by lbarcziova** [comment]
@nforro is right that the messages we currently listen to are filtered in a way that there needs to be mapping to Fedora for the particular package. Besides that, New Hotness does other prefiltering, see [the code](https://github.com/fedora-infra/the-new-hotness/blob/363acd33623dadd5fc3b60a83a528926c7c21fc1/hotness/hotness_consumer.py#L461), e.g. the monitoring enabled in Fedora dist-git, whether the version is newer, whether the package is not retired etc. If we want to support CentOS only packages, we would need to do what Nikola proposes. And in that case, maybe implement the additional checks ourselves.

**Comment by mfocko** [comment]
* keep reacting to `org.fedoraproject.prod.hotness.update.bug.file` for Fedora
* `org.release-monitoring.prod.anitya.project.version.update` for CentOS (and potentially if an RFE occurs for older releases of Fedora)
* keep in mind **not to** react to both for Fedora (i.e. don't create duplicate PRs)
* check the events for potential naming discrepancies between Fedora and CentOS
  * rely on users to make sure they're correct

**Comment by ckelleyRH** [comment]
@mfocko - FYI I am happy to do "real world" battle testing of Packit -> CentOS Stream things as and when it is useful. I am part of the OSCI team responsible for Stream/RHEL and I have some packages in Fedora/Stream/RHEL that I maintain primarily to dogfood the whole packaging experience holistically.
